File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/ReferenceElementDateTime.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/ReferenceElementDateTime.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Datetime;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Schema;

public class ReferenceElementDateTime extends Element
{
	private final Datetime _value;
	private final String _name;
	
	ReferenceElementDateTime(String name, Datetime value)
    {
        this._value = new Datetime(value);
        this._name = name;
    }
	
    public Name name() throws Exception
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.DATE;
    }
    
    protected StringBuilder prettyPrint(int tabIndent) throws Exception
    {
    	return super.prettyPrintHelper(tabIndent, this._value.toString());
    }
    
    public Object getValue()
    {
    	return this._value;
    }

    public Datetime getValueAsDate()
    {
        return this._value;
    }

    public Datetime getValueAsDatetime()
    {
        return this._value;
    }

    public Datetime getValueAsTime()
    {
        return this._value;
    }
}

File: BEmu/ReferenceDataRequest/ReferenceElementDateTime.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="ReferenceDataRequest/ReferenceElementDateTime.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.ReferenceDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class ReferenceElementDateTime : Element
    {
        private readonly Datetime _value;
        private readonly string _name;

        internal ReferenceElementDateTime(string name, DateTime value)
        {
            this._name = name;
            //this._value = new Datetime(value, Datetime.DateTimeTypeEnum.date);
            this._value = new Datetime(value);
        }

        public override Name Name { get { return new Name(this._name); } }
        public override int NumValues { get { return 1; } }
        public override int NumElements { get { return 0; } }
        public override bool IsArray { get { return false; } }
        public override bool IsComplexType { get { return false; } }
        public override Schema.Datatype Datatype { get { return Schema.Datatype.DATE; } }

        internal override StringBuilder PrettyPrint(int tabIndent, bool surroundValueWithQuotes = false)
        {
            return base.PrettyPrintHelper(tabIndent, this._value.ToString());
        }

        public override object GetValue()
        {
            return this._value;
        }

        public override Datetime GetValueAsDate()
        {
            return this._value;
        }

        public override Datetime GetValueAsDatetime()
        {
            return this._value;
        }

        public override Datetime GetValueAsTime()
        {
            return this._value;
        }

    }
}
